### PR TITLE
NIFI-9205 - Update prioritizer configuration

### PR DESCRIPTION
**Pull request by mtien-apache**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

This PR updates the prioritizers configuration on the UI. To test in the UI, create a flow and open the connections configurations. In the settings tab, try the following:
- select a prioritizer by double clicking a prioritizer from the available list
- remove a selected prioritizer, by clicking the "x" icon
- the available prioritizers list is now an ordered list and the order cannot be rearranged

Screen Shot 2021-11-09 at 5 43 40 PM

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by sardell** [comment]
Reviewing now...

**Review by sardell — APPROVED** [review]
@mtien-apache Your changes look good to me. Works as advertised. +1 (non-binding).

**Comment by mcgilman** [comment]
Reviewing...

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-connection-configuration.js [hunk 1]
```diff
@@ -1374,6 +1418,44 @@
...
+
+        /**
+         * Adds the controls to the specified selected draggable element.
+         *
+         * @argument {jQuery} draggableElement
+         */
+        addControlsForSelectedPrioritizer: function (draggableElement) {
+            var removeIcon = $('<div class="draggable-control"><div class="fa fa-remove"></div></div>')
+                .on('click', function () {
+                    // remove the remove ice
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-connection-configuration.js:1444]: Minor typo. Appears to be from copying/pasting from `nf-parameter-contexts`. Can you please update both?

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-connection-configuration.js [hunk 2]
```diff
@@ -1526,7 +1608,13 @@
 
                     // handle each prioritizer
                     $.each(connection.prioritizers, function (i, type) {
-                        $('#prioritizer-available').children('li[id="' + type + '"]').detach().appendTo('#prioritizer-selected');
+                        var selectedPrioritizer = $('#prioritizer-available').children('li[id="' + type + '"]');
+
+                        var draggableElement = selectedPrioritizer.find('div.draggable-control');
+                        if (draggableElement.length === 0 ) {
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-connection-configuration.js:1614]: Is there a scenario where prioritizer from the available list will already have a `draggable-control`?

**Review by mcgilman — COMMENTED** [review]
Thanks for the PR @mtien-apache! Looks great and functions well. Just a couple minor comments/questions.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-connection-configuration.js (hunk 2 above)
Comment (mtien-apache) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-connection-configuration.js]: @mcgilman Thanks for reviewing! This check was accidentally leftover from a previous implementation I had and now a prioritizer from the available list will not have a `draggable-control` anymore. I have removed the check. Can you review again? Thanks!

**Review by mcgilman — APPROVED** [review]
Thanks for the update @mtien-apache! Things look great. Will merge shortly.

**Pull request merged**